src/sentry/api/serializers/models/user.py

```diff
@@ -29,22 +34,91 @@ def manytoone_to_dict(queryset, key, filter_func=None):
     return result
 
 
+class _UserEmails(TypedDict):
+    id: str
+    email: str
+    is_verified: bool
+
+
+class _Organization(TypedDict):
+    slug: str
+    name: str
+
+
+class _Provider(TypedDict):
+    id: str
+    name: str
+
+
+class _Identity(TypedDict):
+    id: str
+    name: str
+    organization: _Organization
+    provider: _Provider
+    dateVerified: str
+    dateSynced: str
+
+
+class _UserSerializerAvatar(TypedDict):
+    avatarType: str
+    avatarUuid: Optional[str]
+
+
+class _UserOptions(TypedDict):
+    theme: str  # TODO: enum/literal for theme options
+    language: str
+    stacktraceOrder: int  # TODO enum/literal
+    timezone: str
+    clock24Hours: bool
+
+
+class UserSerializerResponseOptional(TypedDict, total=False):
+    identities: List[_Identity]
+    avatar: _UserSerializerAvatar
+
+
+class UserSerializerResponse(UserSerializerResponseOptional):
+    id: str
+    name: str
+    username: str
+    email: str
+    avatarUrl: str
+    isActive: bool
+    hasPasswordAuth: bool
+    isManaged: bool
+    dateJoined: str
+    lastLogin: str
+    has2fa: bool
+    lastActive: str
+    isSuperuser: bool
+    isStaff: bool
+    experiments: Dict[str, Any]  # TODO
```

Comment:
experiments also seems like one of those fields we'd want to have "hidden" in API responses. (see TODO in src/sentry/apidocs/spectacular_ports.py)